lib/signed-xml.js

```diff
@@ -399,19 +388,33 @@ SignedXml.prototype.getCanonReferenceXml = function(doc, ref, node) {
   return this.getCanonXml(ref.transforms, node, c14nOptions)
 }
 
-SignedXml.prototype.validateSignatureValue = function(doc) {
+SignedXml.prototype.validateSignatureValue = async function(doc) {
   var signedInfoCanon = this.getCanonSignedInfoXml(doc)
   var signer = this.findSignatureAlgorithm(this.signatureAlgorithm)
-  var res = signer.verifySignature(signedInfoCanon, this.signingKey, this.signatureValue)
-  if (!res) this.validationErrors.push("invalid signature: the signature value " +
-                                        this.signatureValue + " is incorrect")
-  return res
+  return new Promise((resolve) => {
+    signer.verifySignature(signedInfoCanon, this.signingKey, this.signatureValue, (err, res) => {
+      if (err || !res) {
+        this.validationErrors.push("invalid signature: the signature value " +
+            this.signatureValue.toString('base64') + " is incorrect")
```

Comment:
Is it possible for this function to be invoked when `this.signatureValue` is not defined or is null, which would cause `toString` to fail?

Reply:
Yes, but you wouldn't. This function is meant to be an internal function of `checkSignature`.

Possible, yes, based on misuse.